### ISSUE TITLE
Re-install setuptools for python3.8 and python3.9

### ISSUE THREAD

**Issue by Yikun** [dongjoon-hyun/ApacheSparkGitHubActionImage#3]
The latest image [failed to run pyspark test](https://github.com/Yikun/spark/runs/4673344029):
```
ERROR [0.087s]: test_scatter_plot (pyspark.pandas.tests.plot.test_frame_plot_matplotlib.DataFramePlotMatplotlibTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/__w/spark/spark/python/pyspark/pandas/tests/plot/test_frame_plot_matplotlib.py", line 375, in test_scatter_plot
    check_scatter_plot(pdf1, psdf1, x="a", y="b", c="c")
  File "/__w/spark/spark/python/pyspark/pandas/tests/plot/test_frame_plot_matplotlib.py", line 353, in check_scatter_plot
    ax1 = pdf.plot.scatter(x=x, y=y)
  File "/usr/local/lib/python3.9/dist-packages/pandas/plotting/_core.py", line 1636, in scatter
    return self(kind="scatter", x=x, y=y, s=s, c=c, **kwargs)
  File "/usr/local/lib/python3.9/dist-packages/pandas/plotting/_core.py", line 892, in __call__
    plot_backend = _get_plot_backend(kwargs.pop("backend", None))
  File "/usr/local/lib/python3.9/dist-packages/pandas/plotting/_core.py", line 1814, in _get_plot_backend
    module = _load_backend(backend)
  File "/usr/local/lib/python3.9/dist-packages/pandas/plotting/_core.py", line 1746, in _load_backend
    import pkg_resources
ModuleNotFoundError: No module named 'pkg_resources'
```

This problem is due to the PR https://github.com/dongjoon-hyun/ApacheSparkGitHubActionImage/pull/2 (yep, my bad), install the setuptools latest version for pypy3 to [fix build breaks](https://github.com/dongjoon-hyun/ApacheSparkGitHubActionImage/runs/4653344712).

After this, the `setuptools` is **removed from `/usr/lib/python3/dist-packages/`** and re-installed in `/usr/lib/pypy3/dist-packages/`, which was installed by `apt install python-setuptools` and is **also used by `python3.8` and `python3.9`**.

So that we should also do reinstall `setuptools` for `python3.8` and `python3.9` separately.

**Comment by Yikun** [comment]
cc @HyukjinKwon @dongjoon-hyun 

Looks like we should also upgrade the setuptools to latest version like (installed pip and setuptools to latest avaiable version for python3.8 and python3.9, and installed them in /usr/lib/python3.X/dist-packages/):
```
RUN python3.8 -m pip install -U pip setuptools
RUN python3.9 -m pip install -U pip setuptools
```

2021.01.03:
Or just revert pypy3 setuptools now.

**Comment by Yikun** [comment]
BTW, one question, looks like python3.8 is already not used by pyspark test (yes, it used by sbt test, but not docker based, [the python3.8 installed by action directly](https://github.com/apache/spark/blob/master/.github/workflows/build_and_test.yml#L210-L223)), so should we remove python3.8 in here directly if this images is only used by pyspark and sparkr?

**Comment by dongjoon-hyun** [comment]
@Yikun .
1. As you know, we don't use the `latest` image. I built Apache Spark GitHub Action image based on 20211116 as I wrote in the PR, https://github.com/apache/spark/commit/74298042cf6a9273ef9323317014b31fa85c15bc.
2. Your repo's failure mean that your patch, https://github.com/dongjoon-hyun/ApacheSparkGitHubActionImage/commit/aeafd7e05a7d2744651a515c085c372899d37bf4, is incomplete still.

In short, I'll reset the last two commits (mine and yours) because it's not working.

**Comment by dongjoon-hyun** [comment]
The branch is reset to the original and we can re-commit with a working patch.

<img width="387" alt="Screen Shot 2021-12-31 at 11 02 38 AM" src="https://user-images.githubusercontent.com/9700541/147836404-38f0db2a-6aad-4f0b-b5b0-1bee3b066929.png">
.

BTW, we are using a tagged version only. I cleaned up the other images regularly, @Yikun .

**Comment by Yikun** [comment]
@dongjoon-hyun Thanks, I also try to reporduce the bug we met before.

https://github.com/Yikun/ApacheSparkGitHubActionImage/pull/3

But it's not triggered again, I also check it in ARM env, it also works. 

Looks like there are some fixes in upstream, so I guess we can add `coverage` safely now.

**Comment by dongjoon-hyun** [comment]
Thanks. I'll recover the previous two commits, @Yikun .

**Comment by dongjoon-hyun** [comment]
<img width="459" alt="Screen Shot 2022-01-01 at 7 29 59 AM" src="https://user-images.githubusercontent.com/9700541/147854046-17de0a66-ddba-4865-9794-c9325d8f4c8b.png">

**Comment by Yikun** [comment]
@dongjoon-hyun **Only coverage and matplatlib commit is enough, the setuptools upgrade doesn't need anymore** (it was for fix the failed of coverage/matplotlib  commit but this issue not triggered again).

so just keep the your commit is enough at this time, and looks like we should recommit the coverage and matplatlib commit to make the ci retrigger (it would not be failed at this time).

And thanks for your patience.

**Comment by HyukjinKwon** [comment]
@Yikun Quick question: why does it not fail in Apache Spark PRs? Can you make any arbitrary change in pandas API on Spark, and create a PR to trigger the tests in matplitlib?

**Comment by HyukjinKwon** [comment]
If it's because Python 3.8 is not used in PySpark tests, yeah I think we can just remove it in the latest image.

**Comment by dongjoon-hyun** [comment]
@HyukjinKwon . As I wrote on my PR and here, 
- https://github.com/apache/spark/pull/35044
- https://github.com/dongjoon-hyun/ApacheSparkGitHubActionImage/issues/3#issuecomment-1003436280

Apache Spark is using 20211116 + the additional packages only.
>  Please note that I didn't rebuilt the Docker image from the scratch because it fails with some reasons. This image is built based on the previous 20211116 and installed coverage and matplotlib with their transitive dependencies.

**Comment by Yikun** [comment]
@HyukjinKwon , as @dongjoon-hyun mentioned, latest image was not used by spark yet.

And at this point, we still need to **remove pypy3 setuptools line (revert the first commit of ApacheSparkGithubActionImage)**: like https://github.com/dongjoon-hyun/ApacheSparkGitHubActionImage/pull/4. Otherwise the spark master would be failed due to python3.9 lack setuptools. 


as  https://github.com/dongjoon-hyun/ApacheSparkGitHubActionImage/issues/3#issuecomment-1003580954 explained.

**Comment by Yikun** [comment]
> If it's because Python 3.8 is not used in PySpark tests, yeah I think we can just remove it in the latest image.

Thanks, I also tested it, it works in x86 and ARM64 env.
